rpmio/rpmlog.cc

```diff
@@ -412,3 +415,43 @@ void rpmlog (int code, const char *fmt, ...)
 exit:
     errno = saved_errno;
 }
+
+int rpmlogOnce (uint64_t domain, const char * key, int code, const char *fmt, ...)
+{
+    int saved_errno = errno;
+    rpmlogCtx ctx = rpmlogCtxAcquire();
+    int newkey = 0;
+
+    if (ctx) {
```

Comment:
Is this check necessary? We don't seem to be doing it elsewhere in this file (the global `ctx` is always returned).

Reply:
It is not. It is there to limit the scope of the lock. If you want to use rpmlock below to print the numbers for the repeated messages we can't hold a lock there.

Reply:
Don't add a bogus check just to create a scope, if you need a scope then just use a plain { } block.

Reply:
But a cleaner version is actually to wrap the seen-check and the lock into a small seenLog() style helper function.

Reply:
> It is not. It is there to limit the scope of the lock. If you want to use rpmlock below to print the numbers for the repeated messages we can't hold a lock there.

Oh right, that didn't occur to me, thanks for clarifying.